src/helpers.ts

```diff
@@ -0,0 +1,233 @@
+    }
+    // Extract supported RPC calls from the RPC-API section
+    const rpcSectionMatch = mdxContent.match(/<Section title=["']RPC-API["']>[\s\S]*?<Legend \/>[\s\S]*?(\| Method \|[\s\S]*?\|[\s\S]*?\|)(?:\n\n|<\/Section>|$)/);
+    if (rpcSectionMatch) {
+      const tableBlock = rpcSectionMatch[1];
+      // Split into lines, skip header and separator
+      const lines = tableBlock.split('\n').filter(line => line.trim().startsWith('|') && !line.includes(':-----'));
+      if (lines.length > 1) {
+        // Remove header
+        lines.shift();
+        for (const line of lines) {
+          // | `method` | params | rollup | l1 |
+          const cols = line.split('|').map(col => col.trim());
+          if (cols.length >= 5) {
+            // Remove backticks from method
+            const method = cols[1].replace(/`/g, '');
+            const params = cols[2];
+            const rollupBehaviour = cols[3];
+            // Remove <Unsupported />, <Modified />, <Added /> tags from ethereumL1Behaviour
+            let ethereumL1Behaviour = cols[4].replace(/<Unsupported\s*\/>|<Modified\s*\/>|<Added\s*\/>/g, '').trim();
+            supportedRpcCalls.push({ method, params, rollupBehaviour, ethereumL1Behaviour });
+          }
+        }
+      }
+    }
+  } catch (e) {
+    // File not found or parse error, leave as null
+  }
+  return { blockTime, finality, sequencingFrequency, supportedTransactionTypes, gasLimit, messaging, supportedRpcCalls };
+}
+
+export function getRootDir(importMetaUrl: string): string {
+  const __filename = fileURLToPath(importMetaUrl);
+  let rootDir = dirname(__filename);
+  if (rootDir.includes('/build')) {
+    rootDir = rootDir.split('/build')[0];
+  } else if (rootDir.includes('/src')) {
+    rootDir = rootDir.split('/src')[0];
+  }
+  return rootDir;
+}
+
+/**
+ * Lists all rollups by reading .mdx files in the docs directory and extracting title and subtitle.
+ * @param rootDir The root directory of the project
+ * @returns Array of { name, description } objects for each rollup
+ */
+export function listRollupsFromDocs(): Array<{ name: string; description: string }> {
+  const rootDir = getRootDir(import.meta.url);
+  const docsDir = `${rootDir}/RollupCodes/src/docs/`;
+  let rollups: Array<{ name: string; description: string }> = [];
+  try {
+    const files = fs.readdirSync(docsDir);
+    for (const file of files) {
+      if (file.endsWith('.mdx')) {
+        const filePath = path.join(docsDir, file);
+        const content = fs.readFileSync(filePath, 'utf8');
+        const { data } = matter(content);
+        if (data.title && data.subtitle) {
+          rollups.push({ name: data.title, description: data.subtitle });
+        }
+      }
+    }
+  } catch (e) {
+    // Let the caller handle errors if needed
+    throw e;
+  }
+  return rollups;
+}
+
+export const getRollupCodesName = (rollupName: string): string => {
```

Comment:
Does not look good to me. What if we add more rollups? Let's remove the double maintenance from this codebase. We should figure out supported rollups from the submodule